core/tx_cacher.go

```diff
@@ -70,22 +72,31 @@ func (cacher *txSenderCacher) cache() {
 // recover recovers the senders from a batch of transactions and caches them
 // back into the same data structures. There is no validation being done, nor
 // any reaction to invalid signatures. That is up to calling code later.
-func (cacher *txSenderCacher) recover(signer types.Signer, txs []*types.Transaction) {
+func (cacher *txSenderCacher) recover(signer types.Signer, txs []*types.Transaction, stakingTxs []*staking.StakingTransaction) {
 	// If there's nothing to recover, abort
-	if len(txs) == 0 {
+	if len(txs)+len(stakingTxs) == 0 {
 		return
 	}
 	// Ensure we have meaningful task sizes and schedule the recoveries
 	tasks := cacher.threads
-	if len(txs) < tasks*4 {
-		tasks = (len(txs) + 3) / 4
+	if len(txs)+len(stakingTxs) < tasks*4 {
+		tasks = (len(txs) + len(stakingTxs) + 3) / 4
 	}
 	for i := 0; i < tasks; i++ {
-		cacher.tasks <- &txSenderCacherRequest{
-			signer: signer,
-			txs:    txs[i:],
-			inc:    tasks,
+		if i < len(txs) {
```

Comment:
please two separate for loops

Reply:
seems like this file itself is removed from master